pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -461,6 +462,14 @@ func (e *EndpointController) checkLeftoverEndpoints() {
 	}
 	for i := range list.Items {
 		ep := &list.Items[i]
+		if _, ok := ep.Annotations[resourcelock.LeaderElectionRecordAnnotationKey]; ok {
```

Comment:
We discussed that here https://github.com/kubernetes/kubernetes/pull/45478#issuecomment-300267596, I prefer it as well. Happy to switch that over.